src/Core/src/Platform/Android/ActivityLifecycleCallbacks.cs

```diff
@@ -0,0 +1,121 @@
+﻿using System;
+using System.Threading;
+using Android.App;
+using Android.OS;
+using Microsoft.Maui.LifecycleEvents;
+
+namespace Microsoft.Maui
+{
+	class ActivityLifecycleCallbacks : Java.Lang.Object, Application.IActivityLifecycleCallbacks
+	{
+		int startedCounter = 0;
+		int resumedCounter = 0;
+		TimeSpan timeout = TimeSpan.FromMilliseconds(700);
+		Timer? timer;
+		bool pauseSent = true;
+		bool stopSent = true;
+		IServiceProvider services;
+		Application application;
```

Comment:
This is the only part that scares me as we are using this counter that does not (maybe I am wrong) consider services and when applications have no UI. So, if I have an app that has push notifications that pop a toast say, will that get the application pause/resume?

I think the counting part may be best moved to the community toolkit and potentially moved to use the xplat windows or something so all platforms can use it. But even if not xplat, this type seems to mix both the lifecycle events and the activity counting into a single class.

Can we move the counting out of the object and into either a separate object or just use the android lifecycle events as this fires?

Basically, the counting seems very activity specific and we are basically syaing when all activities are minimized, the app is paused - Is that the case (with background services or notofocations/alarms)?

Reply:
How does this work as well when the app is doing a StartActivityForResult and the dev settings is to kill the activity? Will the app report a stop and start while say picking photos? If it really is stopping the app then OK, but if it is just stopping the activity...

Reply:
It's worth noting that android's implementation became much more involved since the last time this code was copied. https://github.com/androidx/androidx/tree/androidx-main/lifecycle/lifecycle-process/src/main/java/androidx/lifecycle
Now it uses content providers and other initialization code to hook-in earlier than any activity.

The lifecycle.process package now depends on lifecycle.runtime package and androidx.startup.startup-runtime package

I guess that NOT taking a dependency also has it's costs like having to maintain the code 😅

Reply:
> It's worth noting that android's implementation became much more involved since the last time this code was copied.

Yes I would go back to depending. Google can improve this functionality

Reply:
It's not such a big package to be afraid of depending on it

Reply:
No, well, if you continue to measure the size of the application by  "Hello World", then this is of course a critical moment